components/ai_chat/core/browser/conversation_handler.cc

```diff
@@ -1056,6 +1066,65 @@ void ConversationHandler::MaybeSeedOrClearSuggestions() {
   }
 }
 
+void ConversationHandler::MaybeFetchOrClearContentStagedConversation() {
+  const bool can_check_for_staged_conversation =
+      ai_chat_service_->HasUserOptedIn() && IsContentAssociationPossible() &&
+      should_send_page_contents_;
+  if (!can_check_for_staged_conversation) {
+    // Clear any staged conversation entries since user might have unassociated
+    // content with this conversation
+    // For now, we assume all staged conversations are 2 entries (question and
+    // answer).
+    if (chat_history_.size() != 2) {
+      return;
+    }
+
+    const auto& last_turn = chat_history_.back();
+    if (last_turn->from_brave_search_SERP) {
+      chat_history_.clear();  // Clear the staged query and summary.
+      OnHistoryUpdate();
+    }
+  }
+
+  // Try later when we get a connected client
+  if (!IsAnyClientConnected()) {
+    return;
+  }
+
+  // Currently only have search query summary at the start of a conversation.
+  if (!chat_history_.empty()) {
+    return;
+  }
+
+  associated_content_delegate_->GetStagedEntriesFromContent(
+      base::BindOnce(&ConversationHandler::OnGetStagedEntriesFromContent,
+                     weak_ptr_factory_.GetWeakPtr()));
+}
+
+void ConversationHandler::OnGetStagedEntriesFromContent(
+    const std::optional<SearchQuerySummary>& search_query_summary) {
+  // Check if all requirements are still met.
+  if (!search_query_summary || !chat_history_.empty() ||
```

Comment:
Should we also check user opt-in here?

Reply:
It's being checked in `MaybeFetchOrClearContentStagedConversation` and there's no way currently to opt-out whilst that's happening, but I'll add the check anyway.